crates/shrs_core/src/history.rs

```diff
@@ -125,9 +127,11 @@ impl History for FileBackedHistory {
     type HistoryItem = String;
 
     fn add(&mut self, item: Self::HistoryItem) {
-        self.hist.insert(0, item);
-        // TODO consider how often we want to flush
-        self.flush().unwrap();
+        if !item.starts_with("history run") {
```

Comment:
The ignore is what allows us to predictably pick the command the user requested. By the point the builtin plugin code runs, the command is already added and so the state of history is unpredictable (or at least, not predictable without know what state history was in before the history run command). I tried to come up with other alternatives because I really don't like doing this either, but came up empty.

If there is any other suggestions, I will gladly change it.

Reply:
Should be fine for now but would be nice to make this more general in future

Reply:
100% agree. I can spend a little bit more time to see if something comes to me.